src/utilities/reactivity.ts

```diff
@@ -0,0 +1,45 @@
+import { isEqual } from 'lodash'
+import { isRef, unref, watch } from 'vue'
+import { isArray, isRecord, mapValues } from '@/utilities'
+
+function getRawValue(value: unknown): unknown {
+  if (typeof value === 'object') {
+    if (isRef(value)) {
+      return unref(value)
+    }
+
+    if (isArray(value)) {
+      return getRawArrayValue(value)
+    }
+
+    if (isRecord(value)) {
+      return getRawRecordValue(value)
+    }
+  }
+
+  return value
+}
+
+function getRawArrayValue(values: unknown[]): unknown[] {
+  return values.map(value => getRawValue(value))
+}
+
+function getRawRecordValue(value: Record<PropertyKey, unknown>): Record<PropertyKey, unknown> {
+  return mapValues(value, (key, value) => getRawValue(value))
+}
+
+export function uniqueValueWatcher(...[source, callback, options]: Parameters<typeof watch>): ReturnType<typeof watch> {
```

Comment:
love this abstraction. Might be worth a composition someday in vue-compositions? `useUniqueValueWatcher`?

Reply:
I thought about that, but its a wrapper around `watch` which isn't a composition. Currently vue-compositions doesn't export any global utilities. So this seemed like the best place for now.